server/src/main/java/org/apache/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -457,6 +470,95 @@ static String getAvaticaConnectionId(Map<String, Object> requestMap)
     return (String) connectionIdObj;
   }
 
+  static String getAvaticaProtobufConnectionId(Service.Request request)
```

Comment:
side note, since we now have avatica stuff as dependencies to this project, it probably makes sense in a follow-up to modify the JSON version to use the types like this is doing instead of deserializing to a map.

Reply:
I'm not sure how much there is to gain from that, what are you thinking would be the advantage? Also, looking through it I can't find an easy to use class like ProtobufTranslationImpl for json.

Reply:
BTW, I guess the reason would be defensive for the most part, to more strongly type the check so that whenever we upgrade the library, stuff would fail at compile time instead of just fail in strange ways at run time if any of the requests ever change, however unlikely. 

[Some of the unit tests for the json path are using the avatica types for json](https://github.com/apache/druid/blob/master/server/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java#L462) to test this area already, so was just thinking since its used as a runtime depend now we could use the expected types for non tests too. But yeah, its definitely not necessary and doesn't provide a lot of gain, nor do I think we should do it in this PR or anything.